temporalcli/commandsmd/commands.md

```diff
@@ -1,221 +1,537 @@
-    * Can have bullets
-      * Each bullet is `* <option-names> (<data-type>) - <short-description>. <extra-attributes>`.
-      * `<option-names>` is `` `--<option-name>` `` and can optionally be followed by ``, `-<short-name>` ``.
-      * `<data-type>` must be one of `bool`, `duration`, `int`, `string`, `string[]`, `string-enum`, `timestamp`, TODO: more
-      * `<short-description>` can be just about anything so long as it doesn't match trailing attributes. Any wrap
-        around to newlines + two-space indention is trimmed to a single space.
-      * `<extra-attributes>` can be:
-        * `Required.` - Marks the option as required.
-        * `Default: <default-value>.` - Sets the default value of the option. No default means zero value of the type.
-        * `Options: <option>, <option>.` - Sets the possible options for a string enum type.
-        * `Env: <env-var>.` - Binds the environment variable to this flag.
-        * ``Alias: `--<alias>`.`` - Sets a flag as an alias of this one. The flag should not be separately defined.
-      * Options should be in order of most commonly used.
-    * Also can have single lines below options that say
-      `Includes options set for [<options-set-name>](#options-set-for-<options-set-link-name>).` which is the equivalent
-      of just having them appended to the bulleted list.
-      * Just because a command may have a couple of similar options with another doesn't mean you _have_ to make a
-        shareable options set. Copy/paste is acceptable.
-* Keep commands in alphabetical order.
-* Commands that have subcommands cannot be run on their own.
-* Keep lines at 120 chars if possible.
-* Add punctuation even at the end of phrases.
+This document is automatically parsed.
+Follow these rules.
+
+IN-HOUSE STYLE
+
+* Wording and grammar.
+  * Run a spell check.
+  * Be clear and concise.
+  * Don't reword the command in the short description. Use distinct supplementary language.
+    * Yes: temporal workflow delete: Remove Workflow Execution
+    * No: temporal workflow delete: Delete Workflow
+  * Re-use and adapt existing wording and phrases wherever possible.
+  * Word command short descriptions as if they began "This command will..."
+    Use sentence casing for the short description.
+  * ID is fully capitalized in text ("the Workflow ID") and Id in [metasyntax](https://en.wikipedia.org/wiki/Metasyntactic_variable) (YourWorkflowId).
+* Avoid parentheticals unless absolutely necessary.
+
+* Wrapping:
+  * Assume a user-visible line length of 80 characters max. `fmt -w 78` can help.
+    * Long descriptions must be pre-wrapped.
+  * When declaring Options follow the wrapping style used elsewhere in this file.
+    Splitting the items into multiple lines improves maintainability with clearer diffs.
+
+* Ordering:
+  * Commands: Use alphabetical order for commands.
+  * Options: Most commonly used options first.
```

Comment:
Can you confirm that the help outputs it that way? Should the docs match the help?

Reply:
Where time allowed, I did as many checks as I could. Sadly time often has not allowed and I haven't had personnel that a job of this complexity might have benefited from. Did the best I could and am happy to have other clear eyes on this.

So: Cannot confirm universally. Did preserve command order.

I have a doc I created of general feedback as I went through this process that may or may not have value. Josh has the link.

Reply:
To clarify what I mean here, is I think option documentation in docs should match what is shown when you run `--help`. But I think it's alphabetical order there. But I think this will be moot when yaml comes about.

Reply:
Oh I see what you're saying. 

Confirmed alpha behavior.

Source:

```
#### Options

* `--frontend-address` (string) -
  Remote endpoint.
  Required.
* `--enable-connection` (bool) -
  Set the connection to "enabled".
```

Output:

```
Usage:
  temporal operator cluster upsert [flags]

Flags:
      --enable-connection         enable cross cluster connection.
      --frontend-address string   IP address to bind the frontend service to.
  -h, --help                      help for upsert

```

Happy to go with whatever fits the right flow and I do have the time tomorrow to change this. If @josh-berry is onboard, I can:

* Remove recs (they're going to go away anyway w/ yaml)
* State "use alphabetical order for all options and commands"
* State "to improve maintainability, put more commonly used options first even though they'll be alphabetized in the output"

From UX perspective, more common first _is_ better ergonomics for users discovery but ¯\_(ツ)_/¯, I also think there should be --help and your pick of --long-help, --detailed-help, --full-help for the "make you scroll back" version we currently emit.

Reply:
hm, unfortunate that option order is not preserved. I think at this point I would prefer we not worry too much about non-user-visible things just so we can get this merged for CLI 1.0.

I do agree with @fairlydurable that being able to sort the most common options first (and/or group them into sections) would be nice. (I'm more used to what Python's `argparse` offers here in preserving order and grouping into sections, but I'm getting the sense Cobra isn't quite that capable.) Not sure if that's something we can consider for the future or if that's just not a thing in Cobra. Regardless, doesn't have to be solved now IMO.

Reply:
Removed section. Easiest solution. Resolving.